src/infra/ConversionTrackingReporter.ts

```diff
@@ -0,0 +1,75 @@
+import { ConversionEvent, COOKIE_PARAM, ListingsClickEvent } from '../models';
+import { DEFAULT_CONVERSION_TRACKING_DOMAIN, LISTINGS_SOURCE_PARAM } from '../models/constants';
+import { CommonConversionData } from '../models/conversiontracking/CommonConversionData';
+import { ConversionTrackingService, HttpRequesterService } from '../services';
+import { calculateSeed } from './CalculateSeed';
+
+const conversionEndpoint = 'conversiontracking/conversion';
+const listingsEndpoint = 'listings';
+
+export class ConversionTrackingReporter implements ConversionTrackingService {
+  private _debug: boolean|undefined;
+  constructor(
+    private httpRequesterService: HttpRequesterService,
+    private debug?: boolean|undefined,
+  ) {
+    this._debug = debug;
+  }
+
+  /**
+   * Prints event details to the console for debugging of analytics events as they fire.
+   * @param event - the details of the event that will be printed
+   * @param type - the type of the event that was tracked
+   */
+  private printEvent(event: string, type: string): void {
+    if (!this._debug) return;
+    console.log(
+      `%c[YextAnalytics]%c- Tracked ${type} event: ${event}`,
+      'background: white; color: blue;',
+      '',
+    );
+  }
+
+  async handleRequest(url: string): Promise<void> {
+    const res = await this.httpRequesterService.get(url.toString());
+    if (res.status !== 200 && !(res.type === 'opaqueredirect' || res.type === 'opaque')) {
+      const errorMessage = await res.text();
+      throw new Error(errorMessage);
+    }
+  }
+
+  private static formatBaseEvent(event: CommonConversionData, params: URLSearchParams): void {
```

Comment:
Are we only dealing with first party cookies for sites?

Reply:
after discussing with @maperalta the `cookieId` is either the first or the third party cookie.  This library doesn't send anything in the headers as it doesn't have direct access to the Document, which is why it seems a bit weird.

If you look at CookieManager in `utils` that has the cookie retrieval logic from ytag and in `yext/pages` I'm using it like this:

```
  enableConversionTracking(): void {
    if (this.canTrack()) {
      this._cookieManager = new CookieManager();
      this._analyticsReporter?.setConversionTrackingEnabled(
        true,
        this._cookieManager.setAndGetYextCookie(),
      );
    }
  }

```

Reply:
let me also tag a WIP PR for yext/pages so you can see the implementation on the front end

Reply:
Here's the proposed usage: https://github.com/yext/pages/pull/151/files#diff-b1eff6c672541942b411af90724e5a14038cd68e1ef1f94333ad26c859c51dc7R80